src/providers/incoming-data/incoming-data.ts

```diff
@@ -848,33 +824,39 @@ export class IncomingDataProvider {
     let requiredFeeRate;
 
     if (payProDetails.requiredFeeRate) {
-      requiredFeeRate =
-        coin === 'eth'
-          ? payProDetails.requiredFeeRate
-          : Math.ceil(payProDetails.requiredFeeRate * 1024);
+      requiredFeeRate = !this.currencyProvider.isUtxoCoin(coin)
+        ? payProDetails.requiredFeeRate
+        : Math.ceil(payProDetails.requiredFeeRate * 1024);
     }
 
-    const stateParams = {
-      amount: payProDetails.amount,
-      toAddress: payProDetails.toAddress,
-      description: payProDetails.memo,
-      data: payProDetails.data,
-      paypro: payProDetails,
-      coin,
-      payProUrl: url,
-      requiredFeeRate
-    };
-    const nextView = {
-      name: 'ConfirmPage',
-      params: stateParams
-    };
-    this.events.publish('IncomingDataRedir', nextView);
-  }
-
-  public getPayProDetails(data: string): Promise<any> {
-    const coin = this.getCoinFromUri(data);
-    const url = this.getPayProUrl(data);
-    let disableLoader = true;
-    return this.payproProvider.getPayProDetails(url, coin, disableLoader);
+    try {
+      const disableLoader = true;
+      const { paymentOptions } = await this.payproProvider.getPayProOptions(
+        url,
+        disableLoader
+      );
+      const { estimatedAmount } = paymentOptions.find(
+        option => option.currency.toLowerCase() === coin
+      );
+      const stateParams = {
+        amount: estimatedAmount,
```

Comment:
ERC20 payprov2 `payProDetails.instructions.amount` returns `0` for amount as the token amount is inside the `payProDetails.instructions[0].data` property. So we use `getPaymentOptions` to "stub" a display amount. Coincidentally the estimatedAmount corresponds to the correct display amount for all coins.